poetry/masonry/utils/tags.py

```diff
@@ -75,11 +75,16 @@ def get_abi_tag(env):
         if get_flag(
             env,
             "Py_DEBUG",
-            lambda: hasattr(sys, "gettotalrefcount"),
-            warn=(impl == "cp"),
+            lambda: (hasattr(sys, "gettotalrefcount") and env.version_info < (3, 8)),
+            warn=(impl == "cp" and env.version_info < (3, 8)),
         ):
             d = "d"
-        if get_flag(env, "WITH_PYMALLOC", lambda: impl == "cp", warn=(impl == "cp")):
+        if get_flag(
+            env,
+            "WITH_PYMALLOC",
+            lambda: (impl == "cp" and env.version_info < (3, 8)),
```

Comment:
Python < 3.8 should be mandatory check instead of fallback value

This line should be changed to:
```python
if env.version_info < (3, 8) and get_flag(...):
```
See reference implementation: https://github.com/pypa/pip/blob/e40d2670facb6cfc815f948c05ca73abb3200f50/src/pip/_vendor/packaging/tags.py#L173-L184

Reply:
That makes sense, although  I don't think `WITH_PYMALLOC` can be overridden by env var as you say. I checked the code of `sysconfig.get_config_vars` (as I was looking for a workaround, while waiting on this fix) and there seems to be no way to actually change the behavior. Which makes this issue even more critical as there really seems to be no workaround (?)

Reply:
I made my previous comment wrong, it can't be changed by env var, but the fallback only takes effect when there is no value returned by `sysconfig.get_config_vars()`, which is not the case for `WITH_PYMALLOC` under `python 3.8`. So with the previous implementation, the value is NOT changed at all.

Reply:
After all, follow the reference implementation used by `pip` would make things right.